### PR TITLE
Coordinate Reference System

### LINKED ISSUE #103: Add temporal extent support for KML

### LINKED ISSUE #109: Output CRS
v.0.5 (#99)
If a file has a CRS different than [4326](https://epsg.io/4326), even though the coordinates are transformed into [4326](https://epsg.io/4326) the output CRS is not 4326. 

Geoextent should return the CRS corresponding to the transformation.

### PR DESCRIPTION
**Pull request by SbastianGarzon**
Fixes #109

- CRS output correspond to the transformation into WGS84. 
- Tests include CRS assessment

Also closes #103, #104 and #105

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -32,66 +32,93 @@ def test_netcdf_extract_bbox():
 def test_kml_extract_bbox():
     result = geoextent.fromFile("tests/testdata/kml/aasee.kml", bbox=True)
     assert "bbox" in result
+    assert "crs" in result
     assert result["bbox"] == pytest.approx([7.594213, 51.942465, 7.618246, 51.957278], abs=tolerance)
+    assert result["crs"] == "4326"
+
+
+@pytest.mark.skipif(sys.platform == "darwin", reason="MacOS does not load the file properly")
+def test_kml_extract_tbox():
```
Comment (SbastianGarzon) [inline tests/test_api.py:41]: KML time extraction was implemented by #99.
This test is included to properly close #103

File: tests/test_cli.py [hunk 2]
```diff
@@ -221,65 +230,72 @@ def test_gml_bbox(script_runner):
...
+                            'tests/testdata/geojson/ausgleichsflaechen_moers.geojson')
     assert ret.success, "process should return success"
     assert ret.stderr == '', "stderr should be empty"
-    assert "[7.6016807556152335, 51.94881477206191, 7.647256851196289, 51.974624029877454]" in ret.stdout, "bboxes and time values of all files inside folder, are printed to console"
-    assert "[6.574722, 51.434444, 4.3175, 53.217222]" in ret.stdout, "bboxes and time values of all files inside folder, are printed to console"
-    assert "[292063.81225905, 5618144.09259115, 302531.3161606, 5631223.82854667]" in ret.stdout, "bboxes and time values of all files inside folder, are printed to console"
+    assert "[7.6016807556152335, 51.94881477206191, 7.647256851196289, 51.974624029877454]" in ret.stdout,\
+        "bboxes and time values of all files inside folder, are printed to console"
+    assert "[6.574722, 51.434444, 4.3175, 53.217222]" in ret.stdout, \
+        "bboxes and time values of all files inside folder, are printed to console"
```
Comment (nuest) [inline tests/test_cli.py:265]: The assert message does not match what is asserted - no time values checked here!

File: tests/test_cli.py [hunk 3]
```diff
@@ -221,65 +230,72 @@ def test_gml_bbox(script_runner):
...
+    assert "[292063.81225905, 5618144.09259115, 302531.3161606, 5631223.82854667]" in ret.stdout, \
```
Comment (nuest) [inline tests/test_cli.py:266]: The assert message does not match what is asserted - no time values checked here!

Also, why would all these coordinates be put out here?

**Review by nuest — CHANGES_REQUESTED** [review]
Had a few comments, but nothing really critical.

I created a couple of issues/comments while thinking about the changes, for later: #111 #112 https://github.com/o2r-project/geoextent/issues/80#issuecomment-754523983

File: geoextent/lib/handleCSV.py [hunk 4]
```diff
@@ -52,32 +52,32 @@ def getBoundingBox(filePath):
...
         else:
-            minlon= (min(spatialLonExtent))
-            maxlon= (max(spatialLonExtent))
-
-        spatialExtent= [minlon,minlat,maxlon,maxlat]
-        if not spatialExtent:
+            minlon = (min(spatialLonExtent))
+            maxlon = (max(spatialLonExtent))
+
+        bbox = [minlon, minlat, maxlon, maxlat]
```
[the remark this answers is not in the document]
Comment (SbastianGarzon) [reply geoextent/lib/handleCSV.py:74]: `handleCSV` is not using GDAL. However it would be a good idea to explore the possibility to use `handleVector` for csv files #114

[the remark this answers is not in the document]
Comment (nuest) [reply geoextent/lib/handleCSV.py:74]: Loading the CSV through GDAL could be a solution - see also #37 

The CSV handler probably tries way to hard to support very rare edge cases anyway!

However, note that I did not say `handleCSV` should use GDAL, but that we need to make sure that the coordinate order we use here matches the output CRS. If the output CRS is fixed - fine. But if the output CRS can be changed by the user #109 then we need to find out how to create the bounding box here.

Long story short: please double check here that the coordinate order is correct for our output CRS and make a source code comment about the fixed coordinate order.

File: geoextent/lib/handleRaster.py [hunk 5]
```diff
@@ -75,7 +75,9 @@ def getBoundingBox(filePath):
         if old_cs.GetAxisMappingStrategy() == 1:
             bbox = [latlongmin[1], latlongmin[0], latlongmax[1], latlongmax[0]]
 
-    return bbox
+    spatialExtent = {"bbox": bbox, "crs": str(hf.WGS84_EPSG_ID)}
```
[the remark this answers is not in the document]
Comment (SbastianGarzon) [reply geoextent/lib/handleRaster.py:78]: Getting the value from new_crs seems like the best option, however due to the differences between GDAL versions I can not access the EPSG code from the osr.SpatialReference() class.

File: geoextent/lib/handleVector.py [hunk 6]
```diff
@@ -75,31 +63,35 @@ def getTemporalExtent(filepath):
...
+                logger.debug("Time feature: {}".format(time_feature))
                 for feat in layer:
                     time = feat.GetField(time_feature)
+                    logger.debug("time {}".format(time))
                     if time is not None:
                         time_list.append(time)
                 layer.ResetReading()
 
                 if len(time_list) != 0:
+                    logger.debug("Time_list: {}".format(time_list))
```
[the remark this answers is not in the document]
Comment (SbastianGarzon) [reply geoextent/lib/handleVector.py:84]: I forgot to delete this, I was using it for debugging. This is a list of all the timestamps detected in the file. I think it is not necessary to add this, but could be useful a message with the length of the list.

File: geoextent/lib/handleVector.py [hunk 7]
```diff
@@ -141,7 +134,11 @@ def getBoundingBox(filepath):
 
     bbox_merge = hf.bbox_merge(geo_dict, filepath)
 
+    if bbox_merge is not None:
```
[the remark this answers is not in the document]
Comment (SbastianGarzon) [reply geoextent/lib/handleVector.py:137]: I need to modify the behavior of ` fromDirectory()`, but it is a good idea !

**Review by nuest — CHANGES_REQUESTED** [review]
Final small change: please document the origin of the test file `ifgi_denkpause.xx`.

Good job addressing all the other comments :rocket: :exclamation:

**Review by nuest — APPROVED** [review]
